--- source/app/forum/child/post/newthread.php
+++ source/app/forum/child/post/newthread.php
@@ -19,13 +19,13 @@
	showmessage('group_nopermission', NULL, ['grouptitle' => $_G['group']['grouptitle']], ['login' => 1]);
 }

-if(!$_G['uid'] && !((!$_G['forum']['postperm'] && $_G['group']['allowpost']) || ($_G['forum']['postperm'] && forumperm($_G['forum']['postperm'])))) {
-	if(!defined('IN_MOBILE')) {
-		showmessage('postperm_login_nopermission', NULL, [], ['login' => 1]);
+if(!((!$_G['forum']['postperm'] && $_G['group']['allowpost']) || ($_G['forum']['postperm'] && forumperm($_G['forum']['postperm'])))) {
+	if(!$_G['uid']) {
+		if(!defined('IN_MOBILE')) {
+			showmessage('postperm_login_nopermission', NULL, [], ['login' => 1]);
+		} else {
+			showmessage('postperm_login_nopermission_mobile', NULL, ['referer' => rawurlencode(dreferer())], ['login' => 1]);
+		}
-	} else {
-		showmessage('postperm_login_nopermission_mobile', NULL, ['referer' => rawurlencode(dreferer())], ['login' => 1]);
-	}
-} elseif(empty($_G['forum']['allowpost'])) {
-	if(!$_G['forum']['postperm'] && !$_G['group']['allowpost']) {
+	} elseif(!$_G['forum']['postperm'] && !$_G['group']['allowpost']) {
		showmessage('postperm_none_nopermission', NULL, [], ['login' => 1]);
	} elseif($_G['forum']['postperm'] && !forumperm($_G['forum']['postperm'])) {
		showmessagenoperm('postperm', $_G['fid'], $_G['forum']['formulaperm']);
	}
 } elseif(empty($_G['forum']['allowpost'])) {
+	if(!$_G['forum']['postperm'] && !$_G['group']['allowpost']) {
+		showmessage('postperm_none_nopermission', NULL, [], ['login' => 1]);
+	} elseif($_G['forum']['postperm'] && !forumperm($_G['forum']['postperm'])) {
+		showmessagenoperm('postperm', $_G['fid'], $_G['forum']['formulaperm']);
+	}
+} elseif($_G['forum']['allowpost'] == -1) {
	showmessage('post_forum_newthread_nopermission', NULL);
 }
